module-4/backend/week_one.md

```diff
@@ -1,20 +1,47 @@
 ## Week One - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's the most useful thing you learned from completing the intermission week work?
+   * Basic Javascript syntax and functionality
+
 2. What are some tools to help debug JavaScript code?
+   * PryJS
+   * Chrome Console
+   * Debugger
+
 3. What are some tools you need in order to unit test your JavaScript?
+   * Mocha and Chai
+
 4. What is the syntax for invoking a function?
+   * someFunction()
+
 5. What's the difference between `==` and `===` in JavaScript?
-6. What's the difference between asynchronous and synchronous JavaScript? 
+   * The === operator does not do a type conversion and the types must be the same to be considered equal
+   * The == operator will compare for equality after doing the necessary type conversion
+
+6. What's the difference between asynchronous and synchronous JavaScript?
+   * asynchronous code will not wait for a line to complete before moving on to the next line whereas synchronous code will wait until the line of code is completed before moving on
+
 7. How do we setup a route when creating an API with Node and Express?
+   * app.get('/api/secrets', function(request, response) {
+   some code here.....
+   }
+
 8. What do we use Knex for?
+   * Knew is a partial ORM thats helps us work with the database. It includes features like migrations and seeding. We can also interact with the database through Knex but using .raw to insert SQL.
+
 9. What's `npm` and what do we use it for?
+   * NPM stands for Node Package Manager and it allows us to install dependencies into our code just as Ruby Gems do in Rails.
 
 #### Review  
 10. What's the MVC design pattern? Describe each part of MVC?
```

Comment:
Great! Also be able to discuss this in non-Ruby terms. How would you explain it without `ERB` or `instance variables`?